packages/core/src/modules/mdoc/__tests__/mdoc.deviceResponse.openid4vp.test.ts

```diff
@@ -0,0 +1,305 @@
+  input_descriptors: [
+    {
+      id: 'org.iso.18013.5.1.mDL',
+      format: {
+        mso_mdoc: {
+          alg: ['EdDSA', 'ES256'],
+        },
+      },
+      constraints: {
+        limit_disclosure: 'required',
+        fields: [
+          {
+            path: ["$['org.iso.18013.5.1']['family_name']"],
+            intent_to_retain: false,
+          },
+          {
+            path: ["$['org.iso.18013.5.1']['given_name']"],
+            intent_to_retain: false,
+          },
+          {
+            path: ["$['org.iso.18013.5.1']['birth_date']"],
+            intent_to_retain: false,
+          },
+          {
+            path: ["$['org.iso.18013.5.1']['issue_date']"],
+            intent_to_retain: false,
+          },
+          {
+            path: ["$['org.iso.18013.5.1']['expiry_date']"],
+            intent_to_retain: false,
+          },
+          {
+            path: ["$['org.iso.18013.5.1']['issuing_country']"],
+            intent_to_retain: false,
+          },
+          {
+            path: ["$['org.iso.18013.5.1']['issuing_authority']"],
+            intent_to_retain: false,
+          },
+          {
+            path: ["$['org.iso.18013.5.1']['issuing_jurisdiction']"],
+            intent_to_retain: false,
+          },
+          {
+            path: ["$['org.iso.18013.5.1']['document_number']"],
+            intent_to_retain: false,
+          },
+          {
+            path: ["$['org.iso.18013.5.1']['portrait']"],
+            intent_to_retain: false,
+          },
+          {
+            path: ["$['org.iso.18013.5.1']['driving_privileges']"],
+            intent_to_retain: false,
+          },
+          {
+            path: ["$['org.iso.18013.5.1']['un_distinguishing_sign']"],
+            intent_to_retain: false,
+          },
+        ],
+      },
+    },
+  ],
+}
+
+describe('mdoc device-response openid4vp test', () => {
+  let deviceResponse: string
+  let mdoc: Mdoc
+  let parsedDocument: Mdoc
+
+  const verifierGeneratedNonce = 'abcdefg'
+  const mdocGeneratedNonce = '123456'
+  const clientId = 'Cq1anPb8vZU5j5C0d7hcsbuJLBpIawUJIDQRi2Ebwb4'
+  const responseUri = 'http://localhost:4000/api/presentation_request/dc8999df-d6ea-4c84-9985-37a8b81a82ec/callback'
+
+  let agent: Agent
+  beforeEach(async () => {
+    agent = new Agent(getInMemoryAgentOptions('mdoc-test-agent', {}))
+    await agent.initialize()
+
+    const devicePrivateAskar = AskarKey.fromJwk({ jwk: Jwk.fromJson(DEVICE_JWK_PRIVATE) })
+    await agent.context.wallet.createKey({
+      keyType: KeyType.P256,
+      privateKey: Buffer.from(devicePrivateAskar.secretBytes),
+    })
+
+    const issuerPrivateAskar = AskarKey.fromJwk({ jwk: Jwk.fromJson(ISSUER_PRIVATE_KEY_JWK) })
+    const issuerPrivateKey = await agent.context.wallet.createKey({
+      keyType: KeyType.P256,
+      privateKey: Buffer.from(issuerPrivateAskar.secretBytes),
+    })
+
+    // this is the ISSUER side
+    {
+      mdoc = await Mdoc.create(agent.context, {
```

Comment:
do we also want to add an oid4vci flow with mdoc? OR lateR?

Reply:
I would do that later. Let's first get the basis merged?